src/Xamarin.Android.Build.Tasks/Tasks/LinkAssembliesNoShrink.cs

```diff
@@ -53,10 +55,18 @@ public override bool RunTask ()
 				for (int i = 0; i < SourceFiles.Length; i++) {
 					var source = SourceFiles [i];
 					var destination = DestinationFiles [i];
+					AssemblyDefinition assemblyDefinition = null;
+
+					if (!MTProfile.IsSdkAssembly (source.ItemSpec) && !MTProfile.IsProductAssembly (source.ItemSpec)) {
+						assemblyDefinition = resolver.GetAssembly (source.ItemSpec);
+						step.CheckAppDomainUsage (assemblyDefinition, (string msg) => Log.LogCodedWarning ("XA2000", msg));
+					}
 
 					// Only run the step on "MonoAndroid" assemblies
 					if (MonoAndroidHelper.IsMonoAndroidAssembly (source) && !MonoAndroidHelper.IsSharedRuntimeAssembly (source.ItemSpec)) {
```

Comment:
At some point I should figure out what the difference is between `Profile.IsSdkAssembly`, `Profile.IsProductAssembly`, and `MonoAndroidHelper.IsSharedRuntimeAssembly`. Maybe one is duplicate?

That is a different PR, though.